etools-searchable-multiselection-menu.html

```diff
@@ -984,6 +1010,77 @@
           if (el) {
             el.invalid = false;
           }
+        },
+
+        _shouldRequestMissingOption: function(url) {
+          if (typeof url === 'string' && url !== '') {
+            return true;
+          }
+          return false;
+        },
+
+        /**
+         * If there are no selected options that are not found in dropdown options then request them from server using
+         * url and ajaxParams(values property will contain missing values)
+         * ajaxParams = {
+         *    values: 1 // single value
+         * }
+         *
+         * ajaxParams = {
+         *    values: 1,2,3,4 // multiple values
+         * }
+         */
+        _requestMissingOptions: function(notFoundValues) {
+          if (!(notFoundValues instanceof Array && notFoundValues.length)) {
+            return;
+          }
+          if (this._shouldRequestMissingOption(this.url)) {
+            var missingOptionsEtoolsAjax = this.$$('#missingOptionsAjax');
+            if (missingOptionsEtoolsAjax) {
+              var notFoundV = notFoundValues.join(',');
+              if (this.ajaxParams && typeof this.ajaxParams === 'object') {
```

Comment:
@robertavram, @ntrncic the endpoint that it's gonna be used for esmm dropdown to get missing options will need to accept `values` GET param(we cound change this name or make it configurable). For example:
* single selection, selected value (1001) is not found in the options, the url will look like this:
```
/some_endpoint_name/?values=1001
``` 
and the request will return a response with a single object that will be included in dropdown options.

*  multiple selection, selected values: 1001, 1002 are not found in the options, the url will look like this:
```
/some_endpoint_name/?values=1001,1002
``` 
and the request will return a response with an array with the missing options objects that will be included in dropdown options.